### ISSUE TITLE
⌨ More action dialog buttons lack context for ATs

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#12218]
## ⌨ Accessibility Issue

On our appliction view for admins, we have a "More actions" sidebar that has a number of buttons you can click to opena dialog and edit the information. The issue is, the context for these buttons relies on adjacent text. When navigating using a screen reader, there is a change this context is missed. Some examples include just hearing the following read out:

- Qualified, button, collapsed dialog
- Two thousand and twenty six, button collapsed dialog
- Not placed, button collapsed

Without the additional context from the visual label, it is not clear what the buttons purpose is.

### WCAG 2.1 Criteria

- [SC 2.4.4 Link Purpose (In Context)](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-refs.html)

## 🦋 Expected Behaviour

The purpose of the buttons can be determined without additional context.

## 📋 Steps to Reproduce

1. Login as admin `admin@test.com`
2. Navigate to an application `/admin/candidates/{candidateId}/application`
3. Open a screen reader
4. Navigate tot he buttons using tab
5. Observe the anouncement lacking context for purpose

## 📸 Screenshot

<img width="244" alt="image" src="https://github.com/user-attachments/assets/a1b08f10-8fc3-48e6-a039-c20809bd8dd1">

## 🙋‍♀️ Proposed Solution

1. Include the label in the button as well
2. Add additional context like "Edit {label} {currentText}"

## ✅ Acceptance Criteria

- [ ] Purpose of action buttons can be determined without additional context
- [ ] Designer approval on any potential styling changes

**Comment by tristan-orourke** [comment]
@Jerryescandon is redesigning this anyway, wants us to do whatever is easiest.

**Comment by esizer** [comment]
So, this is far trickier than I first realized. Either option will not work for the status since it changes based on their status. As an example, when a user has been removed, the status says that but the button is actually to reinstate a removed candidate so doing:

1. "Status: Removed" is not indicate what the button does
2. Adding "Edit status" is okay but then we lose the status for screen readers and it is no entirely accurate :sweat: 

I think for status, maybe we can just update it to no longer be a button and show specific buttons below where "Remove candidate" is in the screen shot for the different available actions? 

The other ones are straight forward and can just have aria labels like:

 - Edit expiry date
 - Edit placement
